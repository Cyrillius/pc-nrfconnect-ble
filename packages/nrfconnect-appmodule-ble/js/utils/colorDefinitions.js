/* Copyright (c) 2016 Nordic Semiconductor. All Rights Reserved.
 *
 * The information contained herein is property of Nordic Semiconductor ASA.
 * Terms and conditions of usage are described in detail in NORDIC
 * SEMICONDUCTOR STANDARD SOFTWARE LICENSE AGREEMENT.
 *
 * Licensees are granted free, non-transferable use of the information. NO
 * WARRANTY of ANY KIND is provided. This heading must NOT be removed from
 * the file.
 *
 */

 'use strict';

import {remote} from 'electron';
import { logger } from '../logging';
import 'nrfconnect-core/css/colordefinitions.less';

const CSS_CLASS_PREFIX = 'colordefinition';
const FALLBACK_COLOR = {r: 255, g: 0, b: 0};

const colors = initColors(findColorDefinitionRules());

function initColors(rules) {
    let colors = {};
    rules.forEach(rule => {
        const regex = new RegExp(`.${CSS_CLASS_PREFIX}-(.*)`);
        let key = regex.exec(rule.selectorText)[1];
        colors[key] = rule.style['color'];
    });
    return colors;
}

function findColorDefinitionRules() {
    let rules = [];
    for (let i = 0; i < document.styleSheets.length; i++) {
        let cssRules = document.styleSheets[i].cssRules;
        for (let j = 0; j < cssRules.length; j++) {
            let rule = cssRules[j];
            if (rule.selectorText && rule.selectorText.indexOf(CSS_CLASS_PREFIX) !== -1) {
                rules.push(rule);
            }
        }
    }
    return rules;
}

export function getColor(color) {
    const rgbString = colors[color];
    if (!rgbString) {
        logger.debug('Color ' + color + ' is not defined');
        return FALLBACK_COLOR;
    }
    const rgbArray = rgbString.replace(/[^\d,]/g, '').split(',');
    return {
        r: rgbArray[0],
        g: rgbArray[1],
        b: rgbArray[2]
    }
}
