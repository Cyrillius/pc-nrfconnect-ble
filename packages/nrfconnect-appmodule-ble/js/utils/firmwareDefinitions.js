/* Copyright (c) 2016 Nordic Semiconductor. All Rights Reserved.
 *
 * The information contained herein is property of Nordic Semiconductor ASA.
 * Terms and conditions of usage are described in detail in NORDIC
 * SEMICONDUCTOR STANDARD SOFTWARE LICENSE AGREEMENT.
 *
 * Licensees are granted free, non-transferable use of the information. NO
 * WARRANTY of ANY KIND is provided. This heading must NOT be removed from
 * the file.
 *
 */

/**
 * Input to pc-nrfjprog-js when updating connectivity firmware. Using the webpack
 * raw-loader to load the content of hex files into the bundle as strings.
 *
 * 0: String containing firmware for nRF51
 * 1: String containing firmware for nRF52
 * filecontent: Boolean specifying if the values of 0 and 1 are actual firmware
 *              content or filesystem paths (true if content, false if paths)
 */

export const firmwareDefinitions115k2 = {
    0: require('raw!pc-ble-driver-js/pc-ble-driver/hex/sd_api_v2/connectivity_1.1.0_115k2_with_s130_2.0.1.hex'),
    1: require('raw!pc-ble-driver-js/pc-ble-driver/hex/sd_api_v3/connectivity_1.1.0_115k2_with_s132_3.0.hex'),
    filecontent: true,
};

export const firmwareDefinitions1m = {
    0: require('raw!pc-ble-driver-js/pc-ble-driver/hex/sd_api_v2/connectivity_1.1.0_1m_with_s130_2.0.1.hex'),
    1: require('raw!pc-ble-driver-js/pc-ble-driver/hex/sd_api_v3/connectivity_1.1.0_1m_with_s132_3.0.hex'),
    filecontent: true,
};
