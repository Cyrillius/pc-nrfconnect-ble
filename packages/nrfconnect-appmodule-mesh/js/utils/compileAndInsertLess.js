/* Copyright (c) 2016 Nordic Semiconductor. All Rights Reserved.
 *
 * The information contained herein is property of Nordic Semiconductor ASA.
 * Terms and conditions of usage are described in detail in NORDIC
 * SEMICONDUCTOR STANDARD SOFTWARE LICENSE AGREEMENT.
 *
 * Licensees are granted free, non-transferable use of the information. NO
 * WARRANTY of ANY KIND is provided. This heading must NOT be removed from
 * the file.
 *
 */

 'use strict';

var fs = require('fs');
var remote = require('electron').remote;
var appPath = remote.getGlobal('appPath');
// var compileLess = require('./js/utils/compileLess.js');
var path = require('path');


//compile less and then insert css/styles.css
function insertStyles(cssPath) {
    var link = document.createElement('link');
    link.setAttribute('rel', 'stylesheet');
    link.setAttribute('type', 'text/css');
    link.setAttribute('href', cssPath);
    document.querySelector('head').appendChild(link);
}

function insertAdditionalCSS(dir) {
    insertStyles(path.join(dir, 'styles.css'));
    insertStyles(path.join(dir, 'resizer.css'));
    insertStyles(path.join(dir, 'react-bootstrap-table-all.min.css'));
}


//TODO Make nrfconnect-appmodule-mesh a dynamic name
var stylesCssPath = path.join(appPath, 'packages', 'nrfconnect-appmodule-mesh',  'css', 'styles.css');
var compileLess = require(path.join(appPath, 'packages', 'nrfconnect-appmodule-mesh',  'js', 'utils', 'compileLess.js'));

fs.stat(stylesCssPath, function (err, stats) {
    if (err || process.env.NODE_ENV !== 'production') {
        compileLess('css/styles.less', 'css/styles.css', function (err) {
            if (err) {
                console.error('Error compiling LESS: ' + err);
            } else {
                console.debug('Successfully compiled LESS');
            }

            insertAdditionalCSS('css');
        });
    } else {
        if (stats.isFile()) {
            // css is pre compiled -> production
            insertAdditionalCSS(path.join(appPath, 'css'));
        } else {
            console.error('Found css file but it is not a file');
        }
    }
});
