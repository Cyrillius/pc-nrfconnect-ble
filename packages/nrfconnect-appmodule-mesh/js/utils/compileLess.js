/* Copyright (c) 2015 Nordic Semiconductor. All Rights Reserved.
 *
 * The information contained herein is property of Nordic Semiconductor ASA.
 * Terms and conditions of usage are described in detail in NORDIC
 * SEMICONDUCTOR STANDARD SOFTWARE LICENSE AGREEMENT.
 *
 * Licensees are granted free, non-transferable use of the information. NO
 * WARRANTY of ANY KIND is provided. This heading must NOT be removed from
 * the file.
 *
 */

'use strict';

var less = require('less');
var fs = require('fs');

module.exports = function compileLess(sourceFile, targetFile, callback) {
    fs.readFile(sourceFile, function (err1, data) {
        if (err1) {
            return callback(err1);
        }

        less.render(data.toString(), function (err2, output) {
            if (err2) {
                return callback(err2);
            }

            fs.writeFile(targetFile, output.css, function (err3) {
                return callback(err3);
            });
        });
    });
};
