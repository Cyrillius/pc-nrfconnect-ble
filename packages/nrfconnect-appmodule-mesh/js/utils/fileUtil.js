/* Copyright (c) 2016 Nordic Semiconductor. All Rights Reserved.
 *
 * The information contained herein is property of Nordic Semiconductor ASA.
 * Terms and conditions of usage are described in detail in NORDIC
 * SEMICONDUCTOR STANDARD SOFTWARE LICENSE AGREEMENT.
 *
 * Licensees are granted free, non-transferable use of the information. NO
 * WARRANTY of ANY KIND is provided. This heading must NOT be removed from
 * the file.
 *
 */

 'use strict';

import fs from 'fs';
import os from 'os';
import {shell} from 'electron';
import childProcess from 'child_process';

export function openFileInDefaultApplication(filePath, callback) {
    fs.exists(filePath, exists => {
        if (!exists) {
            if (callback) {
                callback(new Error(`Could not find file at path: ${filePath}`));
            }

            return;
        }

        const escapedPath = filePath.replace(/ /g, '\\ ');

        // Could not find a method that works on all three platforms:
        // * shell.openItem works on Windows and Linux but not on OSX
        // * childProcess.execSync works on OSX but not on Windows
        if (os.type() === 'Darwin') {
            childProcess.execSync(`open  ${escapedPath}`);
        } else {
            console.log(`want to open ${escapedPath}`)
            const ret = shell.openItem(escapedPath);
            console.log(`shell.openItem() returned ${ret}`);
        }
    });
}
