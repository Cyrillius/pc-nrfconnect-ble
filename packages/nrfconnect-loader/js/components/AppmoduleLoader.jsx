/* Copyright (c) 2016 Nordic Semiconductor. All Rights Reserved.
 *
 * The information contained herein is property of Nordic Semiconductor ASA.
 * Terms and conditions of usage are described in detail in NORDIC
 * SEMICONDUCTOR STANDARD SOFTWARE LICENSE AGREEMENT.
 *
 * Licensees are granted free, non-transferable use of the information. NO
 * WARRANTY of ANY KIND is provided. This heading must NOT be removed from
 * the file.
 *
 */

'use strict';

import '../../css/components/appmodule-loader.less';
import logo from 'nrfconnect-core/resources/nrfconnect_neg.png';

import React, { PropTypes } from 'react';
import { Checkbox } from 'react-bootstrap';
import Immutable from 'immutable';
import AppmoduleList from './AppmoduleList';

export default class AppmoduleLoader extends React.PureComponent {

    constructor(props) {
        super(props);
    }

    static propTypes = {
        appmodules: PropTypes.instanceOf(Immutable.List).isRequired,
        onAppmoduleSelected: PropTypes.func.isRequired
    };

    render() {
        const { appmodules, onAppmoduleSelected } = this.props;

        return (
            <div>
                <div className='appmodule-loader'>
                    <div>
                        <a href='http://www.nordicsemi.com/nRFConnect' target='_blank'>
                            <img className='appmodule-loader-logo' src={logo} />
                        </a>
                    </div>
                    <div className='appmodule-loader-heading'>
                        <h4>nRF Connect</h4>
                    </div>
                </div>
                <div className="panel-body">
                    <AppmoduleList appmodules={appmodules} onAppmoduleSelected={onAppmoduleSelected} />
                    <Checkbox checked={false}>Remember my choice</Checkbox>
                </div>
            </div>
        );
    }
}
