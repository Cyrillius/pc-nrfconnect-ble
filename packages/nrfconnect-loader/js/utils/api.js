import { Record } from 'immutable';

const ImmutableAppmodule = Record({
    name: null,
    version: null,
    title: null,
    description: null,
    author: null,
    icon: null,
    path: null
});

export function getImmutableAppmodule(appmodule) {
    return new ImmutableAppmodule({
        name: appmodule.name,
        version: appmodule.version,
        title: appmodule.title,
        description: appmodule.description,
        author: appmodule.author,
        icon: appmodule.icon,
        path: appmodule.path
    });
}