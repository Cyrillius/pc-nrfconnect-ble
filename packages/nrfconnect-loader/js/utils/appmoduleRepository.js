const fs = require('fs');
const remote = require('electron').remote;

const APP_PATH = getAppPath();

// Get appPath global variable either through electron remote (if
// running in browser), or directly if we are in the main process.
function getAppPath() {
    if (remote) {
        return remote.getGlobal('appPath');
    }
    return global.appPath;
}

function getAppmoduleDirectories() {
    const packageJson = require('../../package.json');
    return packageJson.appmodules;
}

function getAppmodules() {
    const appmoduleDirectories = getAppmoduleDirectories();
    return appmoduleDirectories.map(directory => {
        const contents = fs.readFileSync(`${APP_PATH}/node_modules/${directory}/package.json`);
        return createAppmoduleConfig(JSON.parse(contents));
    });
}

function createAppmoduleConfig(packageJson) {
    return {
        name: packageJson.name,
        description: packageJson.description,
        version: packageJson.version,
        author: packageJson.author,
        title: packageJson.config.title,
        icon: `${APP_PATH}/node_modules/${packageJson.name}/${packageJson.config.icon}`,
    };
}

module.exports = {
    getAppmoduleDirectories,
    getAppmodules,
};
