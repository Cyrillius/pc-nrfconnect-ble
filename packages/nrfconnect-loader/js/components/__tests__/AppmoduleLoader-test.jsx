import React from 'react';
import renderer from 'react-test-renderer';
import { List } from 'immutable';
import { getImmutableAppmodule } from '../../utils/api';
import AppmoduleLoader from '../AppmoduleLoader';

// Have to mock react-dom due to bug: https://github.com/facebook/react/issues/7386
// Can be removed after upgrade to react 15.4.0.
jest.mock('react-dom');

const appmodules = List.of(
    getImmutableAppmodule({
        name: 'nrfconnect-appmodule-ble',
        title: 'ble',
        description: 'ble description',
        icon: 'resources/ble-icon.png'
    }),
    getImmutableAppmodule({
        name: 'nrfconnect-appmodule-mesh',
        title: 'mesh',
        description: 'mesh description',
        icon: 'resources/mesh-icon.png'
    })
);

it('renders correctly', () => {
    const tree = renderer.create(
        <AppmoduleLoader appmodules={appmodules} onAppmoduleSelected={() => {}} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
