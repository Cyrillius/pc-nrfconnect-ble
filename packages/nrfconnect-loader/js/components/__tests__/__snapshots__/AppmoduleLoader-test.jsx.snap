exports[`test renders correctly 1`] = `
<div>
  <div
    className="appmodule-loader">
    <div>
      <a
        href="http://www.nordicsemi.com/nRFConnect"
        target="_blank">
        <img
          className="appmodule-loader-logo"
          src="0de65e70408b48e404a43aef9d424408.png" />
      </a>
    </div>
    <div
      className="appmodule-loader-heading">
      <h4>
        nRF Connect
      </h4>
    </div>
  </div>
  <div
    className="panel-body">
    <div>
      <div
        className="list-group">
        <a
          className="list-group-item"
          href="#"
          onClick={[Function onClick]}>
          <div
            className="appmodule-list-item">
            <div>
              <img
                className="appmodule-icon"
                src="resources/ble-icon.png" />
            </div>
            <div>
              <h4
                className="list-group-item-heading">
                ble
              </h4>
              <p
                className="list-group-item-text">
                ble description
              </p>
            </div>
          </div>
        </a>
        <a
          className="list-group-item"
          href="#"
          onClick={[Function onClick]}>
          <div
            className="appmodule-list-item">
            <div>
              <img
                className="appmodule-icon"
                src="resources/mesh-icon.png" />
            </div>
            <div>
              <h4
                className="list-group-item-heading">
                mesh
              </h4>
              <p
                className="list-group-item-text">
                mesh description
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      className="checkbox"
      style={undefined}>
      <label>
        <input
          checked={false}
          disabled={false}
          type="checkbox" />
        Remember my choice
      </label>
    </div>
  </div>
</div>
`;
